tests/cql/CqlComparisonOperatorsTest.xml

```diff
@@ -93,6 +93,18 @@
 			<expression>Tuple { Id : 1, Name : 'John' } = Tuple { Id : 2, Name : 'John' }</expression>
 			<output>false</output>
 		</test>
+		<test name="TupleEqJohn1John2WithNullName">
+			<expression>Tuple { Id : 1, Name : 'John' } = Tuple { Id : 2, Name : null }</expression>
+			<output>false</output>
```

Comment:
This should be `null`, tuples have no implied ordering, so this is a bug in the Java engine...

Reply:
Ah interesting! I was thinking that null is used to indicate "we don't know if these are equal or not" but in this case, since two of the set fields (Id) are unequal I expected that we could confidently return false, regardless of the status of the other fields. Is this not the case?

Reply:
There's room for interpretation here, but the spec says "and the values for all elements that have values, by name, are equal.", to me, that means the element in question has a value in at least one of the inputs. But I can see it being interpreted as "elements that have values for both inputs", and would allow us to return false here. What I'd be concerned with is making sure that didn't introduce an ordering dependency in tuple comparison, so if we did say it was false, then this should return false too:
```cql
{ Id: null, Name: 'John' } = { Id: null, Name: 'Joe' }
```
And indeed, that is how the Java engine is behaving...

Reply:
And even if we were to long-hand it as a conjunction of ands... that would still return false by 3VL semantics for and. So yes, I think you're right, and the tests as written here are correct. (And the Java engine is behaving correctly). Can we add this test though to make sure we're not introducing an ordering dependency?

Reply:
+1 agreed my 2 cents are this behavior would be nice, as it returns false when two tuples are knowably false!

Makes sense on the ordering dependency. For the Java engine, testing the following actually _does_ return null (with one of the Ids being set and the other unset), so perhaps there is an ordering dep in the Java engine. 

Differing names with one null ID:
```
{ Id: null, Name: 'John' } = { Id: 1, Name: 'Joe' } = null // Java
```

Different IDs with one null name:
```
{ Id: 2, Name: 'John' } = { Id: 1, Name: null } = false // Java
```

So it seems in the current Java implementation there may indeed be a an ordering dependency. 

Current Java engine behavior aside, should we still go ahead with the convention that if any of the fields which are set on both operands are not equal, we should return false? Can add suggested tests cases if we're cool with that! This will mean that engines may need to compare more tuple fields in some cases to make a determination instead of returning null in a greedy fashion. (But false can still be returned in a greedy manner)!

Reply:
Yeah, good catch, there is some ordering dependency there, but yes, I agree we should define it as conjunction of equality of all elements, and an implementation can then be greedy about a false result.

Reply:
Fantastic, sounds good! Added some tests for the order swapped fields.